Add support for EFI, EKF and Generator telemetry

Davidsastresas opened the pull request:
EFI and Generator are already in upstream mavlink common.xml, but I just realized EKF one is ardupilotmega.xml only.

Should we implement something through firmwareplugin to workaround it? Or we should just drop it and not include it?

For the rest of it these changes have been working for some time in a custom build.

DonLakeFlyer commented:
> EFI and Generator are already in upstream mavlink common.xml, but I just realized EKF one is ardupilotmega.xml only.

I made the call years ago that I was going to stop supporting mavlink stuff which was in the ardupilot dialect only. It just makes a big unmanageable mess. It's up to ardupilot to get things into common. So I would drop it.

Davidsastresas commented:
@DonLakeFlyer I understand, it is fair enough.

I rebased to master and I removed EKF, leaving only the EFI and generator, which is in mavlink common.xml.

Thanks!

DonLakeFlyer commented:
Is this good to merge now. It is still marked as Draft.

Davidsastresas commented:
Undrafted and merged, thanks!
